**Introduction à Tensorflow**

Dans ce workshop, on va s'initier à Tensorflow. TensorFlow est une bibliothèque open source de Machine Learning, créée par Google, permettant de développer et d’exécuter des applications de Machine Learning et de Deep Learning. Pour plus d'informations, veuillez vous référer directement sur leur site officiel.

Suite à la sortie officiele de Tensorflow 2.0 récemment, les workshops seront basés sur cette version. Contrairement aux workshops précédents, nous ne vous donnerons pas autant d'indications de code. Le but étant que dès que vous rencontrez une difficulté, vous lancez vos recherches sur le web et si vraiment vous bloquez, vous venez nous voir. Après tout, le web dispose de bien + de ressources que nous. Commençons sans plus tarder !

**1. Opérations de base**

**Exercice:** Définissez 3 tensors constants de type int nommés a, b, c et de valeurs 3, 6 et 8.

Indice: Qu'est ce qu'un tensor ?

In [6]:
import tensorflow as tf

a = tf.constant(2)
b = tf.constant(3)
c = tf.constant(6)

print(a)
print(b)
print(c)

tf.Tensor(2, shape=(), dtype=int32)
tf.Tensor(3, shape=(), dtype=int32)
tf.Tensor(6, shape=(), dtype=int32)


**Résultat attendu**:

tf.Tensor(2, shape=(), dtype=int32)

tf.Tensor(3, shape=(), dtype=int32)

tf.Tensor(6, shape=(), dtype=int32)

**Exercice**: Réalisez les opérations suivantes en tf:
        - a + b
        - c - b
        - b * c
        - a / c
        - c - a * b

In [10]:
add = tf.add(a, b)
sub = tf.subtract(a, b)
mul = tf.multiply(a, b)
div = tf.divide(a, b)
op = tf.subtract(c, tf.multiply(a, b))

print("add =", add.numpy())
print("sub =", sub.numpy())
print("mul =", mul.numpy())
print("div =", div.numpy())
print("op =", op.numpy())

add = 5
sub = -1
mul = 6
div = 0.6666666666666666
op = 0


**Résultat attendu**

add = 5

sub = -1

mul = 6

div = 0.6666666666666666

op = 0

**Exercice**: Calculez la moyenne et la somme de a, b, c définis précedemment.

In [12]:
mean = tf.reduce_mean([a, b, c])
sum = tf.reduce_sum([a, b, c])

print("mean =", mean.numpy())
print("sum =", sum.numpy())

mean = 3
sum = 11


**Résultat attendu**:

mean = 3

sum = 11

**Attends**, la moyenne de 2, 3 et 6 ça ne fait pas 3 ?! Ré-initialisez correctement les variables a, b et c pour obtenir le résultat suivant:

mean = 3.6666667

sum = 11.0

In [13]:
a = tf.constant(2.)
b = tf.constant(3.)
c = tf.constant(6.)

mean = tf.reduce_mean([a, b, c])
sum = tf.reduce_sum([a, b, c])

print("mean =", mean.numpy())
print("sum =", sum.numpy())

mean = 3.6666667
sum = 11.0


On continue, mais cette fois avec des matrices

**Exercice**: Définissez les matrices m1 = [[1., 2.], [3., 4.]] et m2 = [[5., 6.], [7., 8.]] et affichez leur produit

**Indice**: Quelle est la fonction pour multiplier 2 matrices ?

In [15]:
m1 = tf.constant([[1., 2.], [3., 4.]])
m2 = tf.constant([[5., 6.], [7., 8.]])

product = tf.matmul(m1, m2)

print("product =", product.numpy())

product = [[19. 22.]
 [43. 50.]]


**Résultat attendu**:
    
product = [[19. 22.]
 [43. 50.]] 

Félicitations, vous savez maintenant comment réaliser des opérations simple avec Tensorflow. 